### ISSUE TITLE
[Bug] Text dropped as file when allow text not enabled but file allowed

### ISSUE THREAD

**Issue by getBoolean** [alexrabin/FlutterNativeDragAndDrop#8]
When text is not allowed but file is, the text gets dropped as a file.

This might be because of the isDocument(), need to investigate.

**Comment by alexrabin** [comment]
Isn't that how it is supposed to behave? If file is an allowedDataType then all of the other data types will be enabled

**Comment by getBoolean** [comment]
> Isn't that how it is supposed to behave? If file is an allowedDataType then all of the other data types will be enabled

Not text file, just text. I would think plain text shouldn't be categorized as a file right?

**Comment by alexrabin** [comment]
> > Isn't that how it is supposed to behave? If file is an allowedDataType then all of the other data types will be enabled
> 
> Not text file, just text. I would think plain text shouldn't be categorized as a file right?

I think if a file is a plaintext file then it is registered as a file. Plaintext should be categorized as text.

Didn't you make it so when only the file data type is enabled that like pdf files would register as the file datatype. Is this causing the text to be registered this way too?

**Comment by getBoolean** [comment]
> > > Isn't that how it is supposed to behave? If file is an allowedDataType then all of the other data types will be enabled
> > 
> > Not text file, just text. I would think plain text shouldn't be categorized as a file right?
> 
> I think if a file is a plaintext file then it is registered as a file. Plaintext should be categorized as text.
> 
> Didn't you make it so when only the file data type is enabled that like pdf files would register as the file datatype. Is this causing the text to be registered this way too? 

Not a file, I mean when you select text from something like the notes app

**Comment by alexrabin** [comment]
So you set the datatype as file and you dropped in text from the notes app and it registered as a file?

**Comment by getBoolean** [comment]
> So you set the datatype as file and you dropped in text from the notes app and it registered as a file?

Yes

**Comment by alexrabin** [comment]
I still think it has to do with the change you implemented making all data return as a file when datatype file is the only one set. I could be wrong but that's what my gut is telling me. We'll have to look into it further

**Comment by getBoolean** [comment]
> I still think it has to do with the change you implemented making all data return as a file when datatype file is the only one set. I could be wrong but that's what my gut is telling me. We'll have to look into it further

Yes. I believe what is happening is that it doesn't get sent as text because of the shouldAllowText(). It should then be ignored but is instead sent as a file because of the isFile(item) which checks isDocument(item).

We can remove the isDocument(item) check inside isFile(item). Removing the isDocument will mean .numbers files from iCloud won't work, but we can find another solution for that.

As for the shouldAllow for text and url, that was accidentally left in after a revert commit. I'm not sure if we should remove it or not because I'm not sure if doing that will mean urls are categorized as text

That's my theory right now based off my memory. I can test it out later when I get home.

**Comment by alexrabin** [comment]
@getBoolean Currently working on this. Don't remove the isDocument(item) I would prefer that to work then this issue. We'll figure a way around this problem

**Comment by alexrabin** [comment]
@getBoolean Just want to confirm: when you set the only allowed data type to be a file and you drop a string, the string should return with a datatype of text correct?

**Comment by alexrabin** [comment]
I was able to get text and urls to register as their types when only `DropDataType.file` is active:

```swift
else if isFile(item) && !item.itemProvider.canLoadObject(ofClass: NSURL.self)
&& !item.itemProvider.canLoadObject(ofClass: String.self){

...
}
```

**Comment by alexrabin** [comment]
Fixed text from being dropped as a file in #9

**Comment by alexrabin** [comment]
Fixed text data type of strings when file data type is files
